### DISCUSSION TITLE
about the RTDServer in a different project

### DISCUSSION THREAD

**Discussion by SKYC829** [Excel-DNA/ExcelDna#569] [Q&A] [answered]
It seems that rtd server will not work if the rtd server implement in another project?

i have project one to register excel function for call the xlcall.rtd() method, and the project two implement an rtd server,but the excel function calculate finish still no entry the rtd server lifetime.(and i also try use the method ExcelIntegration.RegisterRtdWrapper() to register my rtd server)
the archive is my demo project files

[ExcelRTDTest.zip](https://github.com/Excel-DNA/ExcelDna/files/10784568/ExcelRTDTest.zip)

**Comment by govert** [comment, marked as answer]
You need to add the other assembly into the main .dna file as another ExternalLibrary:

```
<?xml version="1.0" encoding="utf-8"?>
<DnaLibrary Name="ExcelRTDTest Add-In" RuntimeVersion="v4.0" xmlns="http://schemas.excel-dna.net/addin/2020/07/dnalibrary">
  <ExternalLibrary Path="ExcelRTDTest.dll" ExplicitExports="false" LoadFromBytes="true" Pack="true" IncludePdb="false" />
  <ExternalLibrary Path="ExcelRTDTest.Servers.dll" ExplicitExports="false" LoadFromBytes="true" Pack="true" IncludePdb="false" />
</DnaLibrary>
```

It would also make more sense for the other project to only use the "ExcelDna.Integration" package, and not the "ExcelDna.AddIn" package too, as it won't be an add-in itself.

**Reply by SKYC829** [reply]
Thanks! it works ! 👍
